### ISSUE TITLE
require-prop-types raises an error when a validator is present

### ISSUE THREAD

**Issue by danielsamuels** [vuejs/eslint-plugin-vue#713]
**Tell us about your environment**	
* **ESLint version:**	4.19.1
* **eslint-plugin-vue version:**	4.4.0
* **Node version:**	node v8.9.4, yarn 1.4.1

**Please show your full configuration:**

```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "env": {
    "browser": true,
    "commonjs": true,
    "jquery": true
  },
  "extends": [
    "plugin:vue/recommended",
    "eslint:recommended"
  ]
}

```

**What did you do?**	

```vue
<script>
export default {
    props: {
        deviceId: {
            validator: value => value === null || typeof value === Number,
            required: true,
        }
    },
};
</script>

```

**What did you expect to happen?**
According to [the documentation](https://github.com/vuejs/eslint-plugin-vue/blob/4.x/docs/rules/require-prop-types.md) it should be possible to define a `validator` in place of a `type` and have the rule pass.  This is not the case.

**What actually happened?**

```
[ESLint: test.vue]

     Error       4,9: Prop "deviceId" should define at least its type. (vue/require-prop-types)

✗ 1 error, 0 warnings

Double-click on lines to jump to location, [F4] for next, [shift-F4] for previous.

[Finished in 1.0s]
```

**Comment by danielsamuels** [comment]
Looks like this issue is resolved in later versions as of #476.
